server/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java

```
@@ -599,7 +599,13 @@ private IndexGeoPointFieldData fieldData(SearchExecutionContext context) {
                 throw new IllegalArgumentException("failed to find mapper for [" + fieldName + "] for geo distance based sort");
             }
         }
-        return context.getForField(fieldType, MappedFieldType.FielddataOperation.SEARCH);
+        IndexFieldData<?> indexFieldData = context.getForField(fieldType, MappedFieldType.FielddataOperation.SEARCH);
+        if (indexFieldData instanceof IndexGeoPointFieldData) {
```

Is it worth checking indexFieldData for nullity here? If it is null, should we return a proper exception?

---

`instanceOf` does the check already here (i.e. `null instanceOf Any` is `false`)

---

I am not sure how to reproduce that scenario. it is a possible bug in a field type impl perhaps. Yet the case is already covered by the current code. It will throw the exception that I added for null values too.

---

That's correct. My primary focus was on ensuring that an appropriate message exception is returned in that scenario. Catching for nullity, we could return something like
```
 throw new IllegalArgumentException(
            "No index field data found for field [" + fieldName + "] of type [" + fieldType.typeName() + "]"
        );
```